### PR TITLE
First pull

### PR DESCRIPTION
**Pull request by aribornstein**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+RUN git submodule update --init && ./install_deps.sh
+
+# Make NeurboParser
+RUN mkdir -p NeurboParser/build   
+WORKDIR /opt/dynet/NeurboParser/NeurboParser/build
+RUN cmake ..; make -j4
+WORKDIR /opt/dynet/NeurboParser
+
+# install zip python and 
+RUN apt-get update && RUN apt-get install zip -y
```
Comment (berndverst) [inline Dockerfile]: If you want to make the Docker image smaller I recommend combining all the following RUN commands into a single RUN command just like earlier in the Dockerfile.

Comment (berndverst) [reply Dockerfile]: Also, this line is actually wrong. The second `RUN` in-line won't work
Should be `RUN apt-get update && apt-get install zip -y`

Recommend installing all your packages in one command like
`apt-get install package-name1 package-name2 package-name3 -y`

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+RUN apt-get install curl -y
+RUN curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+RUN python get-pip.py
+
+# Install spacy, flask, conllu
+RUN pip install -U spacy && python -m spacy download en
+RUN pip install flask conllu
+
+# Unzip sdp_models
+CMD  unzip /data/pretrained_models/sdp_models.zip && /bin/bash
```
Comment (berndverst) [inline Dockerfile]: Do you need the unzip to happen at run time? Are you mounting this file into the container?

If not, consider moving the unzipping into a previous RUN instruction at Build time.

File: src/SDP_eval.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    # update this with direct pipe from output
+    with open(prediction, 'r') as myfile:
+        data = myfile.read()
+    sentences = parse(data)
+    return json.dumps({"original text":text, 
+                       "parse" : [ob.__dict__ for ob in sentences]})
+
+if __name__ == "__main__":
+    default_nerbo =  "/opt/dynet/NeurboParser/"
```
Comment (berndverst) [inline src/SDP_eval.py]: Should this be `default_neurbo` for consistency?

**Review by berndverst — COMMENTED** [review]
Can you run `pylint` on this? There are some style inconsistencies.

I am not entirely sure what this Neurbo Parser is or how the model training works.

Would be great to have a bit more context in the README, and also explain the motivation behind the Python wrapper.

File: Dockerfile (hunk 2 above)
Comment (berndverst) [reply Dockerfile]: Never mind -- I see that this you map a local folder into the container as `/data/`

File: Dockerfile (hunk 1 above)
Comment (aribornstein) [reply Dockerfile]: Thanks good catch must have forgotten to delete the second RUN will fix that. I'm having trouble with the linter in visual studio code on my machine. Will try to fix that tomorrow I agree that will help things look nicer!

File: src/SDP_eval.py (hunk 3 above)
Comment (aribornstein) [reply src/SDP_eval.py]: yes great catch

**Pull request merged**